### ISSUE TITLE
Unreachable hosts

### ISSUE THREAD

**Issue by pierlauro** [singularityhub/singularity-compose#26]
Hi,
when using `singularity-compose`, the local hostname is overridden and to each containers is assigned a different local ip address. E.g.: on a machine with hostname `localhost` and address `10.1.2.345`, singularity runs a container **xyz** with local ip  `10.0.0.1` and hostname **xyz** .

When executing applications using sockets in containers, however, I noticed it is impossible to reach the endpoints both from outside (from other containers or local machine) and inside (from the same container). Sockets are opened, but they are unreachable.

I also tried running singularity-compose with the `--no-resolv` option, but the issue persists. Any clue why this is happening?

When running containers without singularity-compose, hostname and ip are not overridden and everything works fine.

Thank you

**Comment by vsoch** [comment]
This is unfortunately likely an issue with Singularity - there was a lot of trouble with this first issue when I was developing this library. Could you please provide a dummy case to reproduce your issue? I can try it and if it works for me you’ll need to take this up with Singularity developers.

**Comment by pierlauro** [comment]
I could isolate the problem while using some complex containers, give me some time to write down a more simple example.

**Comment by vsoch** [comment]
Ah found it! This issue has the commands that were useful to debugging core instances: https://github.com/sylabs/singularity/issues/3766

**Comment by pierlauro** [comment]
> Ah found it! This issue has the commands that were useful to debugging core instances: [sylabs/singularity#3766](https://github.com/sylabs/singularity/issues/3766)

Having the same issue! If I start one alpine container with `--network-args`, then its ip is unreachable (even from within the container itself). If I start the container without network options, I am able to ping it from outside. [Obviously I deactivated ICMP on my local machine to test the container one].

Given the issue, would it be possible to configure `singularity-compose` in order to don't use singularity's network options?

**Comment by pierlauro** [comment]
By enabling the debug flag (`singularity-compose -d up`), I can see that the command executed to start a container named **xyz** is: 

```
singularity instance start --bind ./etc.hosts:/etc/hosts --net --network none --network-args "IP=10.22.0.2" --hostname xyz --writable-tmpfs PATH_TO_SIF xyz
```

Trying manually directly with singularity, if the command is executed without `--hostname xyz` everything is working because the network is not virtualized. It definitely seems that `--network none` is ignored if a `hostname` is speficied.

**Comment by pierlauro** [comment]
Solved: #27

**Comment by vsoch** [comment]
Fixed by #27, thanks again @pierlauro !
